### PR TITLE
made Basic UI a configurable service

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer kai@openhab.org

### CODE REVIEW

File: extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    xmlns:config-description="http://eclipse.org/smarthome/schemas/config-description/v1.0.0"
+    xsi:schemaLocation="http://eclipse.org/smarthome/schemas/config-description/v1.0.0
+        http://eclipse.org/smarthome/schemas/config-description-1.0.0.xsd">
+
+  <config-description uri="ui:basic">
+	<parameter name="iconType" type="text" required="true">
+	    <label>Icon Format</label>
+	    <description>Defines whether bitmap or vector icons are used by the UI.</description>
+	    <options>
+	       <option value="png">Bitmap</option>
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml]: You mix (at least in this line) spaces and tabs.

File: extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	</parameter>
+    <parameter name="enableIcons" type="text" required="true">
+        <label>Enable Icons</label>
+        <description>Defines whether UI renders icons for the widgets or not.</description>
+        <options>
+           <option value="true">Enable</option>
+           <option value="false">Disable</option>
+        </options>
+    </parameter>
+    <parameter name="enableIcons" type="text" required="false">
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml]: wrong name (I guess copy and paste error)

File: extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml (hunk 1 above)
Comment (kaikreuzer) [inline extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml]: Indeed - just applied the formatter, now all are tabs (is this something we want to address in future?)

File: extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml (hunk 2 above)
Comment (kaikreuzer) [inline extensions/ui/org.eclipse.smarthome.ui.basic/ESH-INF/config/config.xml]: thanks, indeed a copy&paste error. Fixed now.

**Comment by maggu2810** [comment]
> Indeed - just applied the formatter, now all are tabs (is this something we want to address in future?)

Yes, we should use consistent indentation for XMLs too (tabs or spaces, but be strict which one).

**Pull request merged**